rusk/src/lib/builder/node.rs

```diff
@@ -145,25 +166,30 @@ impl RuskNodeBuilder {
         self
     }
 
+    /// Build the RuskNode and corresponding services
     pub async fn build_and_run(self) -> anyhow::Result<()> {
         let channel_cap = self
             .http
             .as_ref()
             .map(|h| h.ws_event_channel_cap)
             .unwrap_or(1);
         let (rues_sender, rues_receiver) = broadcast::channel(channel_cap);
-
-        let chain_id = self.kadcast.kadcast_id.unwrap_or_default();
+        let (node_sender, node_receiver) = mpsc::channel(1000);
 
         let rusk = Rusk::new(
             self.state_dir,
-            chain_id,
+            self.kadcast.kadcast_id.unwrap_or_default(),
             self.generation_timeout,
             self.gas_per_deploy_byte,
             self.min_deployment_gas_price,
             self.block_gas_limit,
             self.feeder_call_gas,
             rues_sender.clone(),
+            #[cfg(feature = "archive")]
+            self.archive_channel
+                .0
+                .clone()
+                .expect("archive feature set but archive sender not set"),
```

Comment:
AFAIK we're not panicking anywhere in this file, why not return the anyhow error like it was done before?